src/main/java/org/dependencytrack/policy/VersionPolicyEvaluator.java

```diff
@@ -50,25 +53,71 @@ public List<PolicyConditionViolation> evaluate(final Policy policy, final Compon
             return violations;
         }
 
+        
         final var componentVersion = new ComponentVersion(component.getVersion());
+        final var componentPurl = component.getPurl();
+        final var componentPurlType = componentPurl != null ? componentPurl.getType() : null;
+        Version componentVersionObj = null;
+
+        if (componentPurl != null && componentPurlType != null) {
+            componentVersionObj = VersionFactory.forScheme(componentPurlType, component.getVersion());
+        }
```

Comment:
You can simplify this by assuming the `generic` scheme if the component doesn't have a PURL. Then the entire logic can use versatile.